bigtable/client/internal/table_admin.cc

```diff
@@ -193,6 +193,42 @@ bool TableAdmin::CheckConsistency(
   return response.consistent();
 }
 
+std::vector<btproto::Snapshot> TableAdmin::ListSnapshots(
```

Comment:
I wonder if this should be an iterator type.  If we expect this to return thousands to maybe a million entries then a `std::vector<>` (or another collection) is probably ideal.  If we expect millions to billions then we need an iterator.  @ghaisa and @mbrukman thoughts?